rfc/rfc004-distributed-document-format.md

```diff
@@ -0,0 +1,120 @@
+# RFC004 Distributed Document Format 
+
+|  |  |
+| :--- | :--- |
+| Nuts foundation | R.G. Krul |
+| Request for Comments: 004 | Nedap |
+|  | September 2020 |
+
+## Distributed Document Format
+
+### Abstract
+
+### Status of document
+
+This document is currently a draft.
+
+### Copyright Notice
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1.  Introduction
+
+This RFC describes an interoperable, content agnostic data format for distributed networks which provides
+cryptographic authentication and verification of its contents, guaranteed ordering and access control.      
+
+## 2. Terminology
+
+* **Document**: piece of application data enveloped with metadata like signatures and references to other documents
+  published on a distributed network.
+
+Other terminology comes from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Requirements
+The following requirements apply for the document format:
+
+* **Flexible content**: the format must support any kind of payload, textual or binary, large or small.
+* **Consistent ordering**: since documents might revise or require a previous document, there must be ordering that's consistent over the network.
+* **Integrity protected**: since second- or third parties MAY relay documents, document integrity MUST be cryptographically verifiable.
+* **Authenticity protected**: for the same reason as above, document authenticity (who authored the document) MUST be cryptographically verifiable.
+* **Anonymous Confidentiality Control**: since some documents MAY be confidential (just to be read by specific parties)
```

Comment:
This is related to ACL's? I don't think we're going that way. Consents will most likely be evented p2p directly. Are there any other use-cases?

Reply:
Yeah I had trouble spec-ing this anyways since we don't have a PoC or concrete use case for it anyways (at leats, not yet). I'll remove it for now.